### ISSUE TITLE
Bug: ERROR /config/updates.json validation error: IP 2 of 2 is empty for record 

### ISSUE THREAD

**Issue by Narflar** [qdm12/ddns-updater#280]
If a domain from the config never gets an IP, the updates.json stays empty (ip": "",),  If I start ddns-updater it registers that empty IP as an error and shuts down.  I have to delete the updates.json before I restart ddns-updater.

**TLDR**: blank IPs in updates.json crash the app.

1. Is this urgent: No
2. DNS provider(s) you use: Cloudflare / google
3. Program version:  Running version latest built on 2021-12-04T21:55:55Z (commit fa15abb)
4. What are you using to run the container:  docker, Synology
5. Extra information (optional)

Logs:

```
today at 3:21:36 PMRunning version latest built on 2021-12-04T21:55:55Z (commit fa15abb)
today at 3:21:36 PM
today at 3:21:36 PM🔧 Need help? https://github.com/qdm12/ddns-updater/discussions/new
today at 3:21:36 PM🐛 Bug? https://github.com/qdm12/ddns-updater/issues/new
today at 3:21:36 PM✨ New feature? https://github.com/qdm12/ddns-updater/issues/new
today at 3:21:36 PM☕ Discussion? https://github.com/qdm12/ddns-updater/discussions/new
today at 3:21:36 PM💻 Email? quentin.mcgaw@gmail.com
today at 3:21:36 PM💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
today at 3:21:36 PM2021/12/17 15:21:36 ERROR /config/updates.json validation error: IP 2 of 2 is empty for record {"domain":"something.com","host":"@","ips":[{"ip":"0.0.0.0","time":"2021-12-17T13:35:17.276441783-05:00"},{"ip":"","time":"2021-12-17T13:35:17.276441783-05:00"}]}
today at 3:21:36 PM2021/12/17 15:21:36 INFO Shutdown successful
```

Configuration file (**remove your credentials!**):

config.json
```json
{
  "settings": [
    {
      "provider": "google",
      "domain": "something.com",
      "host": "@",
      "username": "username",
      "password": "password",
      "ip_version": "ipv4"
    },
    {
      "provider": "google",
      "domain": "something.com",
      "host": "@",
      "username": "username",
      "password": "password",
      "ip_version": "ipv6"
    }
  ]
}
```

updates.json
```json
{
  "records": [
    {
      "domain": "something.com",
      "host": "@",
      "ips": [
        {
          "ip": "0.0.0.0",
          "time": "2021-12-17T15:22:15.588023297-05:00"
        },
        {
          "ip": "",
          "time": "2021-12-17T15:22:15.588023297-05:00"
        }
      ]
    }
  ]
}
```

Host OS:  Synology 7 DSM

**Comment by guitaristtom** [comment]
I'm also having this issue.

**Comment by scolandrea** [comment]
Same here.

**Comment by semuhphor** [comment]
Ditto...  IP is empty: IP 1 of 1 for record {"domain":"r---com","host":"@","ips":[{"ip":"","time":"2023-11-19T00:28:25.40830184-05:00"}]}

Running in Truenas. 

/mnt/Main/ix-applications/releases/ddns-updater/volumes/ix_volumes/data/config.json
{
  "settings": [
    {
      "domain": "r---.com",
      "host": "@",
      "ip_version": "",
      "key": "---",
      "provider": "godaddy",
      "secret": "---"
    }
  ]
}

On GoDaddy, the IP is set to 127.0.0.1 

admin@truenas[.../ddns-updater/volumes/ix_volumes/data]$ nslookup
> r---.com
;; communications error to 8.8.8.8#53: timed out        <----- Is this the problem? 
Server:         8.8.8.8
Address:        8.8.8.8#53

Non-authoritative answer:
Name:   r---.com
Address: 127.0.0.1
> 

I've replace all of my actual domains with "r---.com"

**Comment by semuhphor** [comment]
Ok .... So I fixed it. 

The problem was in the updates.json file. The IP address in the updates.json file must match the A record in your DNS manager. I had read somewhere that if the A record was set to 127.00.1, it would work, but that's not the case. I reset it to my current IP addres (whatismyipaddres.com) and it works like a champ. 

The file is /mnt/Main/ix-applications/releases/ddns-updater/volumes/ix_volumes/data/updates.json.

The contents are:

{
  "records": [
    {
      "domain": "r---.com",
      "host": "@",
      "ips": [
        {
          "ip": "173.---",
          "time": "2023-11-19T17:56:44.336197227-05:00"
        }
      ]
    }
  ]
}

--- is redacted.

**Comment by qdm12** [comment]
Fixed in 7f0e858fa01ea181fe8898f3fb99631a06ed8ea3

Thank you for your long patience!!
